superchain/superchain-version_test.go

```diff
@@ -0,0 +1,61 @@
+package superchain_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func checkErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will fail if the superchain semver file
+// is not read correctly.
+func TestContractVersionsCheck(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+
+	desiredSemver := superchain.SuperchainSemver.OptimismPortal
+	// TODO complete all members of SuperchainSemver
```

Comment:
This can be achieved either through manually enumerating all of the contracts, or by using reflection to iterate through the members of the struct. Both patterns are in use elsewhere.

Reply:
Because we already have plans to add another contract at the end of this work, I'm leaning towards reflection, then we wouldn't have to come back and edit this test when a new contract gets added - right?

Reply:
Yes. We might want to take a pass through other bits of code that are using the more brittle / handcrafted approach and update them, too.